snuba/query/__init__.py

```diff
@@ -518,3 +519,8 @@ def set_from_clause(self, from_clause: TSimpleDataSource) -> None:
 
     def _eq_functions(self) -> Sequence[str]:
         return tuple(super()._eq_functions()) + ("get_from_clause",)
+
+    def __repr__(self) -> str:
+        from snuba.query.formatters.tracing import format_query
+
+        return "\n".join(format_query(cast(ProcessableQuery[SimpleDataSource], self)))
```

Comment:
Please add a comment on the reason of the cast. Otherwise, reading the code people will wonder why we need that and not know that mypy cannot know that TSimpleDataSource cannot be covariant.